### ISSUE TITLE
Shouldn't we only supply the parameters that require grad to the optimizer?

### ISSUE THREAD

**Issue by xuwangyin** [xuwangyin/dinov2-finetune#1]
Need to double check this https://github.com/xuwangyin/dinov2-finetune/blob/eae26f5634ec7a71a71f2d459cbcd63bef69e782/dinov2_finetune.py#L118

**Comment by ybu-lxd** [comment]
The author uses the VIT trained by DINOv2 to migrate downstream tasks and train a classifier (fully connected layer), instead of training the entire VIT.

**Comment by xuwangyin** [comment]
Yeah, everything is fine. In the code before, the gradient for the feature model was disabled, while the classification head was set to requires_grad

```# Disable gradient for feature model
for param in model.feature_model.parameters():
    param.requires_grad = False
    
for param in model.classifier.parameters():
    param.requires_grad = True```

**Comment by learn01one** [comment]
is this support training the entire VIT?thanks

**Comment by xuwangyin** [comment]
I didn't check, but the following code should work
```
# Disable gradient for feature model
for param in model.feature_model.parameters():
    param.requires_grad = True
    
for param in model.classifier.parameters():
    param.requires_grad = True

# Define loss function
criterion = nn.CrossEntropyLoss()

# Observe that all parameters are being optimized
optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
```
